cpp/arcticdb/version/symbol_list.cpp

```diff
@@ -243,10 +232,25 @@ bool SymbolList::can_update_symbol_list(const std::shared_ptr<Store>& store,
         }
     }
 
-    SymbolList::KeyVector SymbolList::get_all_symbol_list_keys(const std::shared_ptr<StreamSource>& store) const {
+    SymbolList::KeyVector SymbolList::get_all_symbol_list_keys(const std::shared_ptr<StreamSource>& store) {
         std::vector<AtomKey> output;
+        uint64_t uncompacted_keys_found = 0;
         store->iterate_type(KeyType::SYMBOL_LIST, [&] (auto&& key) -> void {
-            output.push_back(to_atom(key));
+            auto atom_key = to_atom(key);
+            if(atom_key.id() != compaction_id) {
+                uncompacted_keys_found++;
+            }
+            if (uncompacted_keys_found == this->max_delta_ * 2 && !warned_expected_slowdown_) {
```

Comment:
Since `warned_expected_slowdown_` is global, you might also want to include "this warning will only appear once" in the warning.

Though I doubt the value of it. I think consistent behaviour is better for this warning.